src/cargo/util/toml/mod.rs

```diff
@@ -2553,6 +2578,29 @@ impl TomlDependency {
                         },
                         TomlDependency::Detailed(d) => {
                             let mut dep = d.clone();
+                            match (
+                                default_features.or(default_features2),
+                                d.default_features.or(d.default_features2)
+                            ) {
+                                // member: default-features = true and
+                                // workspace: default-features = false should turn on
+                                // default-features
+                                (Some(true), Some(false)) => {
+                                    dep.default_features = default_features.or(default_features2);
```

Comment:
I think this might be a little simpler or clearer:

```suggestion
                                    dep.default_features = Some(true);
```